qml/ComboSlider.qml

```diff
@@ -0,0 +1,96 @@
+import QtQuick 2.0
+import QtQuick.Controls 2.0
+import QtQuick.Controls.Material 2.1
+import QtQuick.Layouts 1.12
+
+/* Usage:
+ ComboSlider {
+        id: slider
+        // not needed, used if it exists.
+        defaultValue: "Default!" 
+        model: [strings | numbers]
+        onOptionChanged: {
+            console.log("Value " + value + " at key " + key)
+        }
+    }
+*/
+
+Slider {
+    id: root
+
+    property alias model : internalTicks.model
+    property var defaultValue : null
+
+    signal optionChanged(var key, var value)
+
+    from: 0
+    to: model ? model.length - (defaultValue ? 0 : 1)
+              : 0
+
+    stepSize: 1
+
+    onValueChanged: {
+        if(model.length === value) {
+            optionChanged(-1, defaultValue)
+        } else {
+            optionChanged(value, model[value])
+        }
+    }
+
+    background: Rectangle {
+        x: root.leftPadding
+        y: root.topPadding + root.availableHeight / 2 - height / 2
+        implicitWidth: 200
+        implicitHeight: 1
+        width: root.availableWidth
+        height: implicitHeight
+        color: Material.accent
+
+        Repeater {
+            id: internalTicks
+            model: root.model
+
+            delegate: Rectangle {
+                id: tick
+                width: 1
+                height: parent.implicitHeight * 6
+                x: (index
+                    * ((root.availableWidth - handle.width)
+                        / root.to))
+                    + handle.width / 2
+
+                y: parent.implicitHeight
+                color: Material.accent
+
+                Text {
+                    anchors.top: tick.bottom
+                    id: textDelegate
+                    text: modelData
+                    color: Material.accent
+                    //transform: Rotation { origin.x: 0; origin.y: height / 2; angle: 45}
+                    x: -(width / 2)
+                }
+            }
+        }
+
+        Rectangle {
+            id: defaultTick
+            width: 1
+            height: parent.implicitHeight * 3
+            visible: defaultValue ? true : false
```

Comment:
you should check if `defaultValue == null`, because 0 is also `false` and a valid defaultValue.